### ISSUE TITLE
US-1.3 - Preprocessing Pipeline

### ISSUE THREAD

**Issue by malakzaidi** [malakzaidi/medical-XAI-skin-lesion-classification#14]
**As a ML Engineer, I want a robust preprocessing pipeline to prepare images**
for training.
**Acceptance Criteria:**
- [ ] ISICDataset class created in backend/models/dataset.py
- [ ] Training transforms: resize, crop, flip, rotation, color jitter
- [ ] Validation transforms: resize, normalize
- [ ] ImageNet normalization applied
- [ ] DataLoader created with batching
- [ ] Unit tests for dataset class
- [ ] Documentation of transforms
**Technical Specs:**
- Target size: 224×224
- Augmentation: RandomResizedCrop, HorizontalFlip, Rotation(±15°)
- Normalization: ImageNet stats [0.485, 0.456, 0.406], [0.229, 0.224, 0.22
5]
- Batch size: 32
**Estimated Time:** 1h15
